endpoint.cpp

```diff
@@ -344,6 +344,48 @@ void Endpoint::print_statistics()
     printf("\n}\n");
 }
 
+void Endpoint::untrim_msg(const struct buffer *from, struct buffer *to)
```

Comment:
> 1. write the data you have
> 1. write the missing zeros
> 1. write the remaining data you need, if you do
This is exactly what untrim does. Or you mean write to file? That would make the implementation I little bit more complicated even more changing no non-blocking writing.